projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -88,14 +80,55 @@ export class ActionMenuComponent<R, T> {
         return this._actionDisplayConfig;
     }
 
-    @Input() calculateActionsAvailability: boolean = true;
+    /**
+     * To show or hide {@link ActionType.STATIC_FEATURED} and {@link ActionType.STATIC} actions in a dropdown
+     */
+    get shouldDisplayStaticAndStaticFeaturedActionsDropdown(): boolean {
+        return (
+            this.shouldDisplayStaticActions(this.actionStyling.DROPDOWN) ||
+            this.shouldDisplayStaticFeaturedActions(this.actionStyling.DROPDOWN)
+        );
+    }
+
+    /**
+     * To show or hide {@link ActionType.STATIC_FEATURED} and {@link ActionType.STATIC} actions in a inline action bar
+     */
+    get shouldDisplayStaticAndStaticFeaturedActionsInline(): boolean {
+        return (
+            this.shouldDisplayStaticActions(this.actionStyling.INLINE) &&
+            this.shouldDisplayStaticFeaturedActions(this.actionStyling.INLINE)
+        );
+    }
+
+    /**
+     * When there are no nested actions and if all of the contextual actions are marked to be featured, there is no need to show
+     * the contextual actions dropdown in the inline actions container
+     */
+    shouldDisplayContextualActionsDropdownInline = false;
+
+    private _actionDisplayConfig: ActionDisplayConfig = getDefaultActionDisplayConfig();
+
+    /**
+     * Used for calculating the available actions. So, updating the property should update available actions in the action lists
+     */
+    @Input() set calculateActionsAvailability(val: boolean) {
```

Comment:
A setter with a verb? I can't make sense of it. It's better not to say how it's used, you should explain its meaning. Why does the parent set this flag, this seems internal?

Reply:
I agree that setting `calculateActionsAvailability` from outside looks awkward - what is the case when actions availability should not be calculated?
In addition if I use `[calculateActionsAvailability]="false"` I would expect they are not calculated, but looking in the code the `this.updateActionLists();` will be called

Reply:
Added a separate Jira task to fix this in a different PR in Andromeda release https://jira.eng.vmware.com/browse/VDUCC-323